utf8.h

```diff
@@ -1305,8 +1305,15 @@ typedef enum {
     PL_utf8_to_bytes_use_temporary,
 } Perl_utf8_to_bytes_arg;
 
+/* (bool*) 1 is because this parameter should not be used in this case, but
+ * there is a NN assertion for it.  It causes that to passbut still segfault if
+ * wrongly gets used */
 #define Perl_utf8_to_bytes_overwrite(mTHX, s, l)                            \
-        Perl_utf8_to_bytes_(aTHX_ s, l, PL_utf8_to_bytes_overwrite)
+        Perl_utf8_to_bytes_(aTHX_ s, l, (bool *) 1,                         \
+                                  PL_utf8_to_bytes_overwrite)
```

Comment:
This cast `(bool *) 1` is unsafe.

Such casts are implementation defined, and may result in an invalid pointer, use (which the function call itself does) of which may result in undefined behaviour.

Reply:
I was too lazy to look up how to use INT2PTR, which would have been less effort than what it ended up costing me.  I changed to use that and also to using U8* instead of bool.  Presumably that is no longer UB

Reply:
No, it's still invalid.

Reply:
Does that mean `INT2PTR` is unsafe.  I'm expecting bad things to happen should this ever get exercised.  It may be undefined behavior, but I can't imagine it would be worse than a segfault.

(I worked on a machine once that had an instruction that should have been labelled `BLOW FUSE` instead of what it was supposed to do)

Reply:
INT2PTR() or any cast of an integer to a pointer is unsafe unless that integer is zero, or the result of casting a pointer to some object (or a one past the end pointer) to an integer.

Now, an implementation can allow such casts, and they often do, but it is unsafe going by the standard.

Frankly, if you want to pass a pointer to not-an-object, pass NULL, that is what is it for.

Reply:
"Handles" and "opaque" pointers less than +/- 4096 away from signed integer 0, or +/- 128 or +/- 8 away from zero are not going to ever be de-referenceable values. Opaque pointers -1 and +1  are alignment faults anyways in a court room trial. Harvard architecture, 65KB pointers and segmented memory, or 16-bit Windows OS allowing the user to randomly remove a floppy that has current process/current executing .dll/.so/.exe!!! without a crash!!!!, only a temp popup box.

Its not coming back. And HPUX's NULL is a valid readable no-SEGV address in ISO C isn't coming back either.

Only danger is going from 0x40/-64 or +64 to 0x80  between byte 1 and byte 2, brings up danger of accidental sign extension of your flag, in a sloppy codebase. Throw ontop of that a "clever" `&` to test for pointers instead of <>, and you have a bug ticket.

Only real danger of 21st century is getting negative malloc pointers back from libc. Some Unix RISC CPU/OS companies decided user-mode is always negative malloc pointers https://github.com/Tencent/rapidjson/issues/1596 

And "2 free bytes" with AMD64 in 64-bit mode is just that (+Intel), AMD64 in 64 bit mode. Other older OSes spread out the default per proc categories of VM pages wide enough, you didn't get a solid USHORT16 of bits to use for personal intra-process uses (encrypting pointers, CRC32s, hash key hints, network forwarding path/leg hints).